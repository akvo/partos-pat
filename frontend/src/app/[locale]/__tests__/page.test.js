import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { NextIntlClientProvider } from "next-intl";

import Home from "../page";

const messages = {
  Register: {
    btnCreateAccount: "Create account",
  },
};

describe("HomePage", () => {
  it("renders a default page nextjs", () => {
    render(
      <NextIntlClientProvider locale="en" messages={messages}>
        <Home />
      </NextIntlClientProvider>,
    );

    const nextjsLogo = screen.getByAltText("Next.js Logo");

    expect(nextjsLogo).toBeInTheDocument();
  });

  it("renders correctly & match with the snapshot", () => {
    const { container } = render(
      <NextIntlClientProvider locale="en" messages={messages}>
        <Home />
      </NextIntlClientProvider>,
    );
    expect(container).toMatchSnapshot();
  });
});
