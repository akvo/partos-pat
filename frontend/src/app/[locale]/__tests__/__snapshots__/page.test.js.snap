// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly & match with the snapshot 1`] = `
<div>
  <div
    class="w-full flex flex-col items-center justify-center gap-9 pt-20 mb-4"
  >
    <h1
      class="text-4xl xl:text-5xl text-center font-extra-bold"
    >
      The Power Awareness Tool
    </h1>
    <div
      class="w-full lg:w-10/12 2xl:w-8/12 text-center"
    >
      <p
        class="text-lg xl:text-xl text-dark-10 leading-8"
      >
        The decision to develop the Power Awareness Tool is based on the assumption that if partners have a better understanding of the way power works in the partnership, they will be in a better position to work towards shifting power in accordance with their shared principles.
      </p>
    </div>
    <a
      href="/en/register"
    >
      <button
        class="ant-btn css-dev-only-do-not-override-vn1oyg ant-btn-primary ant-btn-lg w-fit min-w-36"
        type="button"
      >
        <span>
          Create Account
        </span>
      </button>
    </a>
  </div>
</div>
`;
